src/Angor/Shared/WalletOperations.cs

```diff
@@ -48,22 +48,45 @@ public PsbtData CreatePsbtForTransaction(Transaction transaction, AccountInfo ac
 
         changeAddress = changeAddress ?? accountInfo.GetNextChangeReceiveAddress();
 
-        var utxoDataWithPaths = FindOutputsForTransaction((long)transaction.Outputs.Sum(_ => _.Value), accountInfo);
+        var utxoDataWithPaths = FindOutputsForTransaction((long)transaction.Outputs.Sum(_ => _.Value), accountInfo, spendAll);
         var coins = utxoDataWithPaths.Select(u => new Coin(uint256.Parse(u.UtxoData.outpoint.transactionId), (uint)u.UtxoData.outpoint.outputIndex, Money.Satoshis(u.UtxoData.value), Script.FromHex(u.UtxoData.scriptHex))).ToList();
 
         if (!coins.Any())
             throw new ApplicationException("No coins found to fund the transaction.");
 
-        var builder = new TransactionBuilder(network)
-            .AddCoins(coins)
-            .SetChange(BitcoinAddress.Create(changeAddress, network))
-            .ContinueToBuild(transaction) // Add the predefined outputs
-            .SendEstimatedFees(new FeeRate(Money.Satoshis(feeRate)))
-            .CoverTheRest(); // Ensure enough input value covers outputs + fee, adjusting change if needed
+        var clonedTransaction = network.CreateTransaction(transaction.ToHex());
```

Comment:
perhaps it's better to create it as an Nbitcoin transaction so in the last line you can just call ToHex ?